src/RemoteInboxNotifications/DataSetupForProducts.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Handles $data setup for products.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+use \Automattic\WooCommerce\Admin\PluginsProvider\PluginsProvider;
+use \Automattic\WooCommerce\Admin\RemoteInboxNotifications\SpecRunner;
+
+/**
+ * Handles $data setup for products.
+ */
+class DataSetupForProducts {
+	/**
+	 * Initialize the class
+	 */
+	public static function init() {
+		add_action( 'product_page_product_importer', array( __CLASS__, 'run_on_product_importer' ) );
+		add_action( 'transition_post_status', array( __CLASS__, 'run_on_transition_post_status' ), 10, 3 );
+	}
+
+	/**
+	 * Set initial $data values.
+	 *
+	 * @param $data object The $data object.
+	 */
+	public static function init_data( &$data ) {
```

Comment:
What do you think about having this return a new `$data` object instead of modifying by reference? _(Personally I think it makes it more obvious at the call that data mutation is taking place)_